[github-actions] docker build is failing due to mplatform/mquery error

**Describe the bug**

Docker build is failing due to mplatform/mquery error.

**Console/log output**
```
Run docker run --rm mplatform/mquery openthread/environment:latest
Unable to find image 'mplatform/mquery:latest' locally
latest: Pulling from mplatform/mquery
db6020507de3: Pulling fs layer
f11a2bcbeb86: Pulling fs layer
db6020507de3: Verifying Checksum
db6020507de3: Download complete
db6020507de3: Pull complete
f11a2bcbeb86: Verifying Checksum
f11a2bcbeb86: Download complete
f11a2bcbeb86: Pull complete
Digest: sha256:e15189e3d6fbcee8a6ad2ef04c1ec80420ab0fdcf0d70408c0e914af80dfb107
Status: Downloaded newer image for mplatform/mquery:latest
ERROR: Failure code from our HTTP request: 400
Error: Process completed with exit code 1.
```

**Solution**

Replace mplatform/mquery with buildx imagetools inspect.

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -52,8 +52,7 @@ jobs:
     - uses: actions/checkout@v2
       with:
         submodules: true
-    - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v1
+
     - name: Prepare
       id: prepare
       run: |
@@ -72,29 +71,26 @@ jobs:
           --build-arg VCS_REF=${GITHUB_SHA::8} \
           ${TAGS} --file ${DOCKER_FILE} .
 
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
     - name: Docker Buildx (build)
       run: |
         docker buildx build --output "type=image,push=false" ${{ steps.prepare.outputs.buildx_args }}
 
-    - name: Docker Login
+    - name: Login to DockerHub
       if: success() && github.repository == 'openthread/openthread' && github.event_name != 'pull_request'
-      env:
-        DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
-        DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
-      run: |
-        echo "${DOCKER_PASSWORD}" | docker login --username "${DOCKER_USERNAME}" --password-stdin
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
 
     - name: Docker Buildx (push)
       if: success() && github.repository == 'openthread/openthread' && github.event_name != 'pull_request'
       run: |
         docker buildx build --output "type=image,push=true" ${{ steps.prepare.outputs.buildx_args }}
 
-    - name: Docker Check Manifest
+    - name: Inspect Image
       if: always() && github.repository == 'openthread/openthread' && github.event_name != 'pull_request'
       run: |
-        docker run --rm mplatform/mquery ${{ steps.prepare.outputs.docker_image }}:${{ steps.prepare.outputs.version }}
-
-    - name: Clear
-      if: always() && github.repository == 'openthread/openthread' && github.event_name != 'pull_request'
-      run: |
-        rm -f ${HOME}/.docker/config.json
+        docker buildx imagetools inspect ${{ steps.prepare.outputs.docker_image }}:${{ steps.prepare.outputs.version }}

